### PR TITLE
Upgrade to web3py6 and add support for external libraries. Remove brownie.

### PR DESCRIPTION
**Pull request by gnpar**
So far this PR implements support for external libraries with web3py and is working with `web3py>=6`. It also implements a compatibility layer with hardhat's project to use it as building platform. Still pending:

- Fully remove brownie support and dependency
- Fix tests

### CODE REVIEW

File: src/ethproto/w3wrappers.py [hunk 1]
```diff
@@ -204,16 +221,24 @@ def __init__(self, receipt, contract):
 
     @property
     def events(self):
+        import ipdb; ipdb.set_trace()  # fmt: skip
```
Comment (gnarvaja) [inline src/ethproto/w3wrappers.py:224]: Y esto?

File: tests/hardhat-project/artifacts/contracts/CounterWithLibrary.sol/CounterWithLibrary.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
+{
+  "_format": "hh-sol-artifact-1",
```
Comment (gnarvaja) [inline tests/hardhat-project/artifacts/contracts/CounterWithLibrary.sol/CounterWithLibrary.json:2]: Porqué subir todos los artifacts al repo?

File: tests/test_contract_with_library.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class CounterUpgradeableWithLibrary(Counter):
+    eth_contract = "CounterUpgradeableWithLibrary"
+    proxy_kind = "uups"
+    constructor_args = ()
+    initialize_args = (("initial_value", "int"),)
+
+
+def test_deploy_counter():
```
Comment (gnarvaja) [inline tests/test_contract_with_library.py:29]: No podrías usar tests parametrizados? Según lo que entiendo, todas las implementaciones del contrato tienen el mismo funcionamiento.

File: src/ethproto/w3wrappers.py [hunk 4]
```diff
@@ -255,29 +359,31 @@ def eth_function(*args):
...
 
     def _handle_exception(self, err):
         if str(err).startswith("execution reverted: "):
-            raise RevertError(str(err)[len("execution reverted: "):])
+            raise RevertError(str(err)[len("execution reverted: ") :])
         super()._handle_exception(err)
 
+    def _get_msg_args(self):
+        # TODO: This needs to be zero for test runs, but forcing it to zero like this will break other clients
+        return {"gasPrice": 0}
```
Comment (gnpar) [inline src/ethproto/w3wrappers.py:372]: Shall we merge this as is and then find a better solution?

Ideally this would be handled at the provider level or on a pytest fixture for tests.

**Review by gnarvaja — APPROVED** [review]
Aprobado, salvo porque no me queda claro el impacto del gasPrice:0 en _get_msg_args cuando se quiera usar esta librería no para tests sino para interactuar con la blockchain.
